packages/sdk-middleware-auth/src/base-auth-flow.js

```diff
@@ -92,7 +100,8 @@ function executeRequest(
       })
     })
     .catch((error: Error) => {
-      response.reject(error)
+      if (response && typeof response.reject === 'function')
```

Comment:
With `node-fetch` it seems like you do not always have to now can reject. The tests showed this which is why I added the guard.

Reply:
The `response.reject` I think is from the SDK middleware and it should always be defined

https://github.com/commercetools/nodejs/blob/d7fce290e3ff5c65b59637ee5b79faefad73e06f/packages/sdk-client/src/client.js#L66-L72

So I'm not sure this is necessary...

Reply:
I tried again to remove it. It leads to "ensure to fetch new token only once and keep track of pending tasks" to fail. 

It's an integration test which was running on Node.js with `isomorphic-fetch` having pulled in the `whatwg-fetch` polyfill before. Now it runs with `node-fetch`. I would like to keep the guard.

Regarding the "Initial response shape": I didn't touch that bit and would like to keep it if we're both not sure what it's for.